v3/pkg/application/systemtray_windows.go

```diff
@@ -64,25 +57,45 @@ func (s *windowsSystemTray) positionWindow(window *WebviewWindow, offset int) er
 	newX := screenBounds.Width - window.Width()
 	newY := screenBounds.Height - window.Height()
 
+	// systray icons in windows can either be in the taskbar
+	// or in a flyout menu.
+	iconIsInTrayBounds, err := s.iconIsInTrayBounds()
+	if err != nil {
+		return err
+	}
+
+	// we only need the traybounds if the icon is in the tray
+	var trayBounds *Rect
+	if iconIsInTrayBounds {
+		trayBounds, err = s.bounds()
+		if err != nil {
+			return err
+		}
+	}
+
 	taskbarBounds := w32.GetTaskbarPosition()
+
+	// Set the window position based on the icon location
+	// if the icon is in the taskbar (traybounds) then we need
+	// to adjust the position so the window is centered on the icon
 	switch taskbarBounds.UEdge {
 	case w32.ABE_LEFT:
-		if trayBounds != nil && trayBounds.Y-(window.Height()/2) >= 0 {
+		if iconIsInTrayBounds && trayBounds.Y-(window.Height()/2) >= 0 {
 			newY = trayBounds.Y - (window.Height() / 2)
 		}
 		window.SetRelativePosition(offset, newY)
 	case w32.ABE_TOP:
-		if trayBounds != nil && trayBounds.X-(window.Width()/2) <= newX {
+		if iconIsInTrayBounds && trayBounds.X-(window.Width()/2) <= newX {
 			newX = trayBounds.X - (window.Width() / 2)
 		}
 		window.SetRelativePosition(newX, offset)
 	case w32.ABE_RIGHT:
-		if trayBounds != nil && trayBounds.Y-(window.Height()/2) <= newY {
+		if iconIsInTrayBounds && trayBounds.Y-(window.Height()/2) <= newY {
 			newY = trayBounds.Y - (window.Height() / 2)
 		}
 		window.SetRelativePosition(screenBounds.Width-window.Width()-offset, newY)
 	case w32.ABE_BOTTOM:
-		if trayBounds != nil && trayBounds.X-(window.Width()/2) <= newX {
+		if iconIsInTrayBounds && trayBounds.X-(window.Width()/2) <= newX {
```

Comment:
All of these cases could be handled by a single check above for !iconIsInTrayBounds and bailing

Reply:
Yeah, we could wrap the switch in an if. You ok with that @tw1nk ? 🙏